src/opentime/timeRange.h

```diff
@@ -45,57 +54,199 @@ class TimeRange {
+        double otherEnd = other.end_time_exclusive().to_seconds();
+        return fabs(otherStart - selfStart) <= epsilon && otherEnd - selfEnd >= epsilon;
+    }
+
+    /**begins(other, epsilon)
+     * params: RationalTime other
+     * The start of self strictly equals other.
+     */
+    bool begins(RationalTime other, double epsilon = defaultEpsilon_s) const {
+        double selfStart = _start_time.to_seconds();
+        double otherStart = other.to_seconds();
+        return fabs(otherStart - selfStart) <= epsilon;
+    }
+
+    /**finishes(other, epsilon)
+     * params: TimeRange other
+     * The start of self strictly antecedes the start of other by a value >= epsilon.
+     * The end of self strictly equals the end of other.
+     * The converse would be other.finishes(self)
+     */
+    bool finishes(TimeRange other, double epsilon = defaultEpsilon_s) const {
+        double selfStart = _start_time.to_seconds();
+        double selfEnd = end_time_exclusive().to_seconds();
+        double otherStart = other._start_time.to_seconds();
+        double otherEnd = other.end_time_exclusive().to_seconds();
+        return fabs(selfEnd - otherEnd) <= epsilon && selfStart - otherStart >= epsilon;
+    }
+
+    /**finishes(other)
+     * params: RationalTime other
+     * The end of self strictly equals other.
+     */
+    bool finishes(RationalTime other, double epsilon = defaultEpsilon_s) const {
+        double selfEnd = end_time_exclusive().to_seconds();
+        double otherEnd = other.to_seconds();
+        return fabs(selfEnd - otherEnd) <= epsilon;
+    }
+
+
+    /**operator equals()
+     * params: TimeRange lhs
+     *         TimeRange rhs
+     * The start of lhs strictly equals the start of rhs.
+     * The end of lhs strictly equals the end of rhs.
+     */
+    friend bool operator==(TimeRange lhs, TimeRange rhs) {
+        RationalTime start = lhs._start_time - rhs._start_time;
+        RationalTime duration = lhs._duration - rhs._duration;
+        return fabs(start.to_seconds()) < defaultEpsilon_s &&
+               fabs(duration.to_seconds()) < defaultEpsilon_s;
+    }
+
+    /**operator notequals()
+     * params: TimeRange lhs
+     *         TimeRange rhs
+     * Converse of equals() operator
+     */
+    friend bool operator!=(TimeRange lhs, TimeRange rhs) {
         return !(lhs == rhs);
     }
 
     static TimeRange range_from_start_end_time(RationalTime start_time, RationalTime end_time_exclusive) {
-        return TimeRange {start_time, RationalTime::duration_from_start_end_time(start_time, end_time_exclusive)};
+        return TimeRange{start_time,
+                         RationalTime::duration_from_start_end_time(start_time, end_time_exclusive)};
+    }
+
+    static constexpr double getDefaultEpsilon(){
+        return defaultEpsilon_s;
     }
 
 private:
     RationalTime _start_time, _duration;
+    static constexpr double defaultEpsilon_s = 0.5 / 192000.0;
```

Comment:
Two things about this:

1)
```suggestion
    static constexpr double defaultEpsilon_s = 1 / (2*192000.0);
```
This is more clear about the value being the inverse of double 192k.

2) Lets add a comment before it:
```This default epsilon value is used in comparison between floating numbers.  It is computed to be twice 192khz, the fastest commonly used audio rate.  It can be changed in the future if necessary due to higher sampling rates or some other kind of numeric tolerance detected in the library.```